src/unix/linux-core.c

```diff
@@ -1228,6 +1228,57 @@ uint64_t uv_get_constrained_memory(void) {
 }
 
 
+static uint64_t uv__get_available_memory_fallback(void) {
+  uint64_t constrained;
+  uint64_t total;
+  uint64_t current;
+
+  constrained = uv__get_constrained_memory_fallback();
+
+  total = uv_get_total_memory();
+  if (constrained == 0 || constrained > total)
+    return uv_get_free_memory();
+
+  current = uv__read_uint64("/sys/fs/cgroup/memory/memory.usage_in_bytes");
+  return constrained - current;
+}
+
+
+uint64_t uv_get_available_memory(void) {
+  char filename[4097];
+  char buf[1024];
+  uint64_t constrained;
+  uint64_t total;
+  uint64_t current;
+  char* p;
+
+  constrained = uv_get_constrained_memory();
+  if (constrained == 0)
+    return uv__get_available_memory_fallback();
+
+  if (uv__slurp("/proc/self/cgroup", buf, sizeof(buf)))
+    return uv__get_available_memory_fallback();
+
+  if (memcmp(buf, "0::/", 4))
+    return uv__get_available_memory_fallback();
+
+  total = uv_get_total_memory();
+  if (constrained > total)
+    return uv_get_free_memory();
```

Comment:
I was going to suggest moving this code up a few lines so you don't read & parse /proc/self/cgroup for nothing but I realize that's not 100% equivalent because these lines won't be reached unless we're on a cgroupsv2 system.

That's fairly subtle assuming it's intentional so please at least add a comment explaining the what and why.